JEDI/SOCA cap modifications for T,S,U, and V

mark-a-potts opened the pull request:
Exposes the full 3d fields for T, S, U, and V to SOCA when built with the UFS in NG-GODAS or S2S mode.

jiandewang commented:
@mark-a-potts : can you combine your commit history into one. The general rule for MOM6 is not to use squash in merging but developers using as less commit history as possible. You can do force push from now on when you need to do further modification. Thanks

jiandewang commented on config_src/drivers/nuopc_cap/mom_cap_methods.F90:
```diff
@@ -352,14 +354,15 @@ subroutine mom_export(ocean_public, ocean_grid, ocean_state, exportState, clock,
   integer                         :: day, secs
   type(ESMF_TimeInterval)         :: timeStep
-  integer                         :: dt_int
+  integer                         :: dt_int,ii,jj
```
need to add space between them

mark-a-potts replied:
They were actually unused, so I removed them.

mark-a-potts commented:
> 

Sorry about that. I squashed all the changes since merging to dev/emc into one and renamed it, but the list of commits are still showing here. I can re-do the PR with a clean push, if you would like.

jiandewang commented:
> > 
> 
> Sorry about that. I squashed all the changes since merging to dev/emc into one and renamed it, but the list of commits are still showing here. I can re-do the PR with a clean push, if you would like.

if you can re-do that in a clean push, that will be great. Thanks.
